### ISSUE TITLE
[Feature Request] Target components with dcc.Loading

### ISSUE THREAD

**Issue by ndrezn** [plotly/dash#1922]
**Is your feature request related to a problem? Please describe.**
Generally, it can be frustrating that `dcc.Loading` will trigger a spinning animation when ANY of its nested components are in a callback. For instance, you might want a full-screen spinner to take over sometimes, but if you want a full-screen spinner takeover, that animation is going to run EVERY time that a callback in the app runs.... which is quite painful.

**Describe the solution you'd like**
New properties for `dcc.Loading`, e.g. `property=` and/or `component_id=` (or similar) which make it possible to target specific nested component properties and component IDs.

The code to handle this would likely go here: https://github.com/plotly/dash/blob/dev/components/dash-core-components/src/components/Loading.react.js#L54-L57

And look something like:
```js
if (loading_state.prop != this.props.property) {
     return this.props.children
}
```

**Describe alternatives you've considered**
There isn't a good solution here. You can down-nest the loading spinner, but this might not be the right UX you're looking for in the app.

**Comment by CNFeffery** [comment]
@ndrezn you can try the `AntdSpin()` in my components library [fac](https://github.com/CNFeffery/feffery-antd-components), set parameter `listenPropsMode='include'`, then every target defined in `includeProps` will not trigger Loading animation, here is a simple example:

```python

fac.AntdSpace(
    [
        fac.AntdButton('in includeProps', id='spin-include-demo-input1', type='primary'),
        fac.AntdButton('not in includeProps', id='spin-include-demo-input2', type='primary')
    ]
),

fac.AntdSpin(
    fac.AntdSpace(
        [
            fac.AntdText('not in includeProps nClicks: 0', id='spin-include-demo-output1', strong=True),
            fac.AntdText('in includeProps nClicks: 0', id='spin-include-demo-output2', strong=True)
        ],
        direction='vertical'
    ),
    listenPropsMode='include',
    includeProps=['spin-include-demo-output2.children'],
    text='running'
)
...
@app.callback(
    Output('spin-include-demo-output1', 'children'),
    Input('spin-include-demo-input1', 'nClicks'),
    prevent_initial_call=True
)
def spin_include_callback_demo1(nClicks):
    import time
    time.sleep(1)

    return f'not in includeProps nClicks: {nClicks}'


@app.callback(
    Output('spin-include-demo-output2', 'children'),
    Input('spin-include-demo-input2', 'nClicks'),
    prevent_initial_call=True
)
def spin_include_callback_demo2(nClicks):
    import time
    time.sleep(1)

    return f'in includeProps nClicks: {nClicks}'
```

**Comment by Coding-with-Adam** [comment]
closed by #2760
